### ISSUE TITLE
How to conquer the Multiverse

### ISSUE THREAD

**Issue by maxdevjs** [maxdevjs/awseome-301DaysOfCode-test#1]
As a quick start, my idea is:

- basically a data/repo where awesome people participating to 301DaysOfCode can
    - add awesome personal data (immortalizing their participation)
    - add awesome resources that turned out to be useful

hopefully with a very low overhead for single maintainers ( [github organization](https://github.com/blog/674-introducing-organizations) ? )

Very simple, very straightforward. 

**Every suggestion will be very welcome**.

Is Github the perfect fit? Currently I can think of one issue, ranking scenario for resources:

- someone adds a book (video/bla/bla), how to know how many people find it actually useful? 
    - counter showing it? How the counter would work? Github pages allow to write some data? Firebase? 
    - opening and linking to an issue? etc
    - who cares about ranking?
    - ...

**Comment by benjaminspak** [comment]
I was thinking of this issue the other day. People enjoy ranking, but how do we make an objective ranking system that favors quality over popular opinion. 

Ultimately, while not automated, we should make this process vetted with certain quality criteria. Someone who shares a resource must have a reason behind why they are sharing it. Then, a small team of reviewers can assess if the resource is good.

**Comment by maxdevjs** [comment]
Interesting points. I share your POV, anyway a couple considerations:

- can we have an objective ranking system? How can we assert that my criteria as reviewer is most valuable of your as proponent? 
- or that quality over popular opinion is the best option when people is targeted beneficiary?
- how to put together such like-minded team?
- ranking is actually needed (from the beginning)?
- from [contributing.md](https://github.com/max-devjs/awseome-301DaysOfCode-test/blob/master/.github/contributing.md) (which is just an example):

```
Use the following format: [Resource Title](link): Description.
```

the description could include the reason. Every submission (and every self adding to the community) would be a pull request, then the team would unfailingly assess the content, if necessary rejecting it.

Then

- a team is needed
- how to maintain everything the simplest possible

**Comment by benjaminspak** [comment]
Hmm good points and questions. 

We could allow members to list:
- overall experience with programming 
- top 3 favorite resources

Then, simply make a list of the members' top 3 resources per member.

**Comment by benjaminspak** [comment]
This would:

- Require members to rank their best/favorite resources.
- We would sacrifice a little objective quality, but not require a team.
- Member would feel like they had greater control.

- We imply the difficulty level of the resources based on the years of experience a member has.
- Then, members would know which resources may be more tailored to beginners, and which may require more of an expert's POV to understand.
- We could also limit the possibility of one member sharing many resources. Which, over time I would assume the quality of their resources would decrease. 
- Also, early adopters could post often, and build up a sub-faction within the group. This would pull focus away from the group as a whole, and onto one member. Ideally, I'd like to avoid this.

**Comment by maxdevjs** [comment]
Something like this?

```
- your surname and name [about](http://example.com/about) | [301-days-code journal](http://example.com/301-days-code-journal) | [twitter](https://twitter.com/example) | [github](https://github.com/example)
- one sentence overall experience with programming when entering 301-days-code (otherwise this would quickly get outdated needing further updates)
- three preferred resources (assuming people would want to be added at the beginning, how can they know what is actually effective for personal learning in the field?)
```

would read:

- your surname and name [about](http://example.com/about) | [301-days-code journal](http://example.com/301-days-code-journal) | [twitter](https://twitter.com/example) | [github](https://github.com/example)
- one sentence overall experience with programming when entering 301-days-code (otherwise this would quickly get outdated needing further updates)
- three preferred resources (people at the beginning can know what is effective for personal learning?)

without a ```learn``` section / folder / entry. Straightforward and manageable, I like it.

At first I thought that the learn section would not require a rigid "commitment", allowing people to feel more free. The three resources structure overtime could make people willing to change the initial idea -> further overload for team / maintainer(s)?

The following are pertinent points:

- We could also limit the possibility of one member sharing many resources.
- Also, early adopters could post often, and build up a sub-faction within the group.

I can be be wrong, but those issues would be naturally addressed by the pull request acceptation process: assumed that the pull request is analyzed by someone you trust, automatically (human fault allowing) some skimming is embedded and inevitable. 

Paradoxically, early adopters would be the easier to be checked, due to smaller number of entries

**Comment by benjaminspak** [comment]
Hmm that is a really good point. Early adopters would be easier to be checked.  They would have more helpful resources than a newbie would too. 

How about we allow 3 resources as a founder perk? 3 resources for the first 100 members, for example.

**Comment by benjaminspak** [comment]
Maybe the first 10 adapters can contribute more. Then we can narrow the allow the submissions as we grow. If there are dibplicated of resources we can then do a contest to allow member to vote on a new unmentioned resource.

**Comment by benjaminspak** [comment]
Also, you're right. Pull requests would already act as a filter :)

**Comment by maxdevjs** [comment]
Yes, good idea (resources).

Twitter Polls (contest)?

**Comment by benjaminspak** [comment]
Yeah, but I'm open to more ideas :)

**Comment by maxdevjs** [comment]
Summarizing, what could be the structure? Everything in readme?

**Comment by benjaminspak** [comment]
Do you think a readme would be the best medium? Or should we list the info on a webpage?

**Comment by maxdevjs** [comment]
I think Github needs a more effective notification system :smiley_cat: 

Sorry for the late replies.

I do not know. The readme could be a fair starting point, probably would quickly become a mess. If not enough, better to have b and c plans already available. 

Anyway, as soon as possible I will start my 300etc journey. I probably will maintain a journal of it. Thinking of a double source system: .md (as backup and readable on a normal repo) and "transpiled" to html pages (for Github pages...?). I will put something in a repo in a couple days, may be can be a base for further ideas.

In the meantime, do you have a particular structure in mind?

**Comment by benjaminspak** [comment]
So true haha 

I think what we came up with before will work well. And I agree, MD will be a good starting point. :)

**Comment by maxdevjs** [comment]
This has been dormant for some time now. What you think, may have some usefulness?

**Comment by benjaminspak** [comment]
Yeah, we can use it as a template, if you're like to build out 301daysofcode.com together :)

**Comment by maxdevjs** [comment]
Oh, wait... switching context, tell me about it :)

Edit.: how do you think it could be implemented?

**Comment by max-devjs** [comment]
Oh, wait... switching context, tell me about it :)

**Comment by max-devjs** [comment]
Oh, wait... switching context, tell me about it :)
